### PR TITLE
[TRUS-3805] - Legally Incapable, Mapper and Model Updates

### PR DESCRIPTION
**Pull request by jassalrichy**
Legally Incapable Controller, View, Page & Unit Specs. Updated Individual Beneficiary Model, Mapper and Unit Specs

### CODE REVIEW

File: app/models/beneficiaries/IndividualBeneficiary.scala [hunk 1]
```diff
@@ -40,32 +43,42 @@ object IndividualBeneficiary extends BeneficiaryReads {
...
       __.lazyRead(readNullableAtSubPath[IndividualIdentification](__ \ 'identification)) and
       __.lazyRead(readNullableAtSubPath[Address](__ \ 'identification \ 'address)) and
-      (__ \ 'vulnerableBeneficiary).read[Boolean] and
+      (__ \ 'vulnerableBeneficiary).readNullable[Boolean] and
       (__ \ 'beneficiaryType).readNullable[RoleInCompany] and
       (__ \ 'beneficiaryShareOfIncome).readNullable[String] and
       (__ \ 'beneficiaryDiscretion).readNullable[Boolean] and
+      (__ \ 'countryOfResidence).readNullable[String] and
+      (__ \ 'nationality).readNullable[String] and
+      (__ \ 'legallyIncapable).readNullable[Boolean] and
```
Comment (jackheslop96) [inline app/models/beneficiaries/IndividualBeneficiary.scala:52]: I think we can change this to `(__ \ 'mantalCapacity).readNullable[Boolean].map(_.map(!_))` then this will give you `legallyIncapable` and you won't have to call `.map(!_)` in each case

File: app/models/beneficiaries/IndividualBeneficiary.scala [hunk 2]
```diff
@@ -40,32 +43,42 @@ object IndividualBeneficiary extends BeneficiaryReads {
...
       (__ \ 'identification \ 'address).writeNullable[Address] and
-      (__ \ 'vulnerableBeneficiary).write[Boolean] and
+      (__ \ 'vulnerableBeneficiary).writeNullable[Boolean] and
       (__ \ 'beneficiaryType).writeNullable[RoleInCompany] and
       (__ \ 'beneficiaryShareOfIncome).writeNullable[String] and
-      (__ \ 'beneficiaryDiscretion).write[Boolean] and
+      (__ \ 'beneficiaryDiscretion).writeNullable[Boolean] and
+      (__ \ 'countryOfResidence).writeNullable[String] and
+      (__ \ 'nationality).writeNullable[String] and
+      (__ \ 'legallyIncapable).writeNullable[Boolean] and
```
Comment (jackheslop96) [inline app/models/beneficiaries/IndividualBeneficiary.scala:77]: So I _think_ we can make this `(__ \ 'mentalCapacity).writeNullable[Boolean](Writes[Boolean](x => JsBoolean(!x)))` and then revert the changes below

File: app/models/beneficiaries/IndividualBeneficiary.scala (hunk 1 above)
Comment (jassalrichy) [reply app/models/beneficiaries/IndividualBeneficiary.scala]: Updated

File: app/models/beneficiaries/IndividualBeneficiary.scala (hunk 2 above)
Comment (jassalrichy) [reply app/models/beneficiaries/IndividualBeneficiary.scala]: Updated

File: app/utils/mappers/IndividualBeneficiaryMapper.scala [hunk 3]
```diff
@@ -102,9 +124,10 @@ class IndividualBeneficiaryMapper extends Mapper[IndividualBeneficiary]  {
   }
 
   private def readIncome: Reads[Option[String]] = {
-    IncomeDiscretionYesNoPage.path.read[Boolean].flatMap[Option[String]] {
-      case true => Reads(_ => JsSuccess(None))
-      case false => IncomePercentagePage.path.read[Int].map(value => Some(value.toString))
+    IncomeDiscretionYesNoPage.path.readNullable[Boolean].flatMap[Option[String]] {
+      case Some(true) => Reads(_ => JsSuccess(None))
```
Comment (jackheslop96) [inline app/utils/mappers/IndividualBeneficiaryMapper.scala:128]: Can remove this line

File: app/utils/mappers/IndividualBeneficiaryMapper.scala [hunk 4]
```diff
@@ -85,12 +106,13 @@ class IndividualBeneficiaryMapper extends Mapper[IndividualBeneficiary]  {
   }
 
   private def readAddress: Reads[Option[Address]] = {
-    NationalInsuranceNumberYesNoPage.path.read[Boolean].flatMap {
-      case true => Reads(_ => JsSuccess(None))
-      case false => AddressYesNoPage.path.read[Boolean].flatMap[Option[Address]] {
+    NationalInsuranceNumberYesNoPage.path.readNullable[Boolean].flatMap {
+      case Some(true) => Reads(_ => JsSuccess(None))
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/utils/mappers/IndividualBeneficiaryMapper.scala]: Removed

File: app/utils/mappers/IndividualBeneficiaryMapper.scala (hunk 3 above)
Comment (jassalrichy) [reply app/utils/mappers/IndividualBeneficiaryMapper.scala]: Removed

**Review by jackheslop96 — APPROVED** [review]
